src/ledger/LedgerTxn.cpp

```diff
@@ -2242,6 +2253,9 @@ uint32_t
 LedgerTxnRoot::Impl::prefetch(std::unordered_set<LedgerKey> const& keys)
 {
     uint32_t total = 0;
+    auto prefetchTimer =
```

Comment:
Is the reason the code is written like this to make it "exception safe" when not in TRACE mode?
It looks a bit cludgy: it actually performs a bunch of computation that we never use even if we're not in TRACE mode.
Maybe this is related to what I was asking in https://github.com/stellar/stellar-core/pull/2386#issuecomment-584435308
Basically, maybe this (unsafe logging) should only be something that we can enable on test builds.